features/base/conference/middleware.js

```diff
@@ -0,0 +1,79 @@
+import JitsiMeetJS from '../lib-jitsi-meet';
+import { MiddlewareRegistry } from '../redux';
+import {
+    TRACK_ADDED,
+    TRACK_REMOVED
+} from '../tracks';
+
+const JitsiTrackErrors = JitsiMeetJS.errors.track;
+
+/**
+ * This middleware intercepts TRACK_ADDED and TRACK_REMOVED actions to sync
+ * conference's local tracks with local tracks in state.
+ *
+ * @param {Store} store - Redux store.
+ * @returns {Function}
+ */
+const conferenceMiddleware = store => next => action => {
+    let actionType = action.type;
+    let track = action.track;
+
+    if ((actionType === TRACK_ADDED || actionType === TRACK_REMOVED) &&
+        track &&
+        track.isLocal()) {
+        return syncConferenceLocalTracksWithState(store, action)
+            .then(() => next(action));
+    } else {
+        return next(action);
+    }
+};
+
+/**
+ * Syncs local tracks from state with local tracks in JitsiConference instance.
+ *
+ * @param {Store} store - Redux store.
+ * @param {Object} action - Action object.
+ * @returns {Promise}
+ */
+const syncConferenceLocalTracksWithState = (store, action) => {
```

Comment:
I find `function f(store, action) {...}` to be easier to read than `const f = (store, action) => {...}`. I think another advantage that I've read about is that the Function object is named instead of anonymous which is handy during debugging. (Again, I'm merely noting & not calling for any action.)